### PR TITLE
Feat: Assign conversations to teams.

### PR DESCRIPTION
**Pull request by nithindavid**
Todo
- Translations
- Warnings on team switch

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue [hunk 1]
```diff
@@ -159,6 +198,46 @@ export default {
...
+      set(item) {
+        const agentId = item ? item.id : 0;
+        this.$store.dispatch('setCurrentChatAssignee', item);
+        this.$store
+          .dispatch('assignAgent', {
+            conversationId: this.currentChat.id,
+            agentId,
+          })
+          .then(() => {
+            bus.$emit('newToastMessage', this.$t('CONVERSATION.CHANGE_AGENT'));
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:220]: Please use alert mixin

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue [hunk 2]
```diff
@@ -159,6 +198,46 @@ export default {
...
+    },
+    teamsList() {
+      return [{ id: 0, name: 'None' }, ...this.teams];
+    },
+    assignedAgent: {
+      get() {
+        return this.currentChat.meta.assignee;
+      },
+      set(item) {
+        const agentId = item ? item.id : 0;
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:212]: item.id || 0

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue [hunk 3]
```diff
@@ -159,6 +198,46 @@ export default {
...
+      set(item) {
+        const teamId = item ? item.id : 0;
+        this.$store.dispatch('setCurrentChatTeam', item);
+        this.$store
+          .dispatch('assignTeam', {
+            conversationId: this.currentChat.id,
+            teamId,
+          })
+          .then(() => {
+            bus.$emit('newToastMessage', this.$t('CONVERSATION.CHANGE_TEAM'));
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:237]: Please use alertMixin

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue [hunk 4]
```diff
@@ -159,6 +198,46 @@ export default {
...
+            bus.$emit('newToastMessage', this.$t('CONVERSATION.CHANGE_AGENT'));
+          });
+      },
+    },
+    assignedTeam: {
+      get() {
+        return this.currentChat.meta.team;
+      },
+      set(item) {
+        const teamId = item ? item.id : 0;
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:229]: imte.id || 0

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue [hunk 5]
```diff
@@ -159,6 +198,46 @@ export default {
...
+        this.$store.dispatch('setCurrentChatTeam', item);
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:230]: Please use a better variable name than item.

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue [hunk 6]
```diff
@@ -159,6 +198,46 @@ export default {
...
+    teamsList() {
+      return [{ id: 0, name: 'None' }, ...this.teams];
+    },
+    assignedAgent: {
+      get() {
+        return this.currentChat.meta.assignee;
+      },
+      set(item) {
+        const agentId = item ? item.id : 0;
+        this.$store.dispatch('setCurrentChatAssignee', item);
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:213]: Please use a better name than item.

File: app/javascript/dashboard/store/modules/conversations/index.js [hunk 7]
```diff
@@ -145,7 +150,7 @@ export const mutations = {
   // Update assignee on action cable message
   [types.default.UPDATE_ASSIGNEE](_state, payload) {
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/conversations/index.js:151]: ASSIGN_AGENT is similar to UPDATE_ASSIGNEE right? Can we merge these two?

File: app/views/api/v1/accounts/conversations/messages/index.json.jbuilder [hunk 8]
```diff
@@ -3,6 +3,7 @@ json.meta do
   json.additional_attributes @conversation.additional_attributes
   json.contact @conversation.contact.push_event_data
   json.assignee @conversation.assignee.push_event_data if @conversation.assignee.present?
+  json.assignee @conversation.assignee.push_event_data if @conversation.assignee.present?
```
Comment (pranavrajs) [inline app/views/api/v1/accounts/conversations/messages/index.json.jbuilder:6]: duplicate. Did you mean json.team?

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please look at the individual comments.

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue (hunk 2 above)
Comment (nithindavid) [reply app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:212]: Item returned is null when unselected.

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue (hunk 4 above)
Comment (nithindavid) [reply app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:229]: Item returned is null when unselected.

File: app/javascript/dashboard/store/modules/conversations/index.js (hunk 7 above)
Comment (nithindavid) [reply app/javascript/dashboard/store/modules/conversations/index.js:151]: It takes a refactoring to change it.

File: app/views/api/v1/accounts/conversations/messages/index.json.jbuilder (hunk 8 above)
Comment (nithindavid) [reply app/views/api/v1/accounts/conversations/messages/index.json.jbuilder:6]: @sojan-official Do we need team data here ?

**Pull request merged**